syntax = "proto3";

package zense;

message Image {
  uint32 width = 1;
  uint32 height = 2;
  uint32 channel = 3;
  uint64 timestamp = 4;
  int32 depth_range = 5;
  bytes data = 6;
}

service ZenseService{
  rpc SendRGBDImage (ImageRequest) returns (ImageRGBDReply) {}
  rpc SendRGBDIRImage (ImageRequest) returns (ImageRGBDIRReply) {}  
  rpc SendDepthIRImage (ImageRequest) returns (ImageDepthIRReply) {}
  rpc SendWDRDepthImage (ImageRequest) returns (ImageWDRDepthReply) {}
}

message ImageRequest{
}

message ImageRGBDReply{
  Image image_rgb = 1;
  Image image_depth = 2;
}

message ImageDepthIRReply{
  Image image_ir = 1;
  Image image_depth = 2;
}

message ImageRGBDIRReply{
  Image image_rgb = 1;
  Image image_ir = 2;
  Image image_depth = 3;
}

message ImageWDRDepthReply{
  Image image_depth_range1 = 1;
  Image image_depth_range2 = 2;
}
